cmd/thanos/store.go

```diff
@@ -84,6 +84,11 @@ func registerStore(m map[string]setupFunc, app *kingpin.Application) {
 	consistencyDelay := modelDuration(cmd.Flag("consistency-delay", "Minimum age of all blocks before they are being read. Set it to safe value (e.g 30m) if your object storage is eventually consistent. GCS and S3 are (roughly) strongly consistent.").
 		Default("0s"))
 
+	ignoreDeletionMarksDelay := modelDuration(cmd.Flag("ignore-deletion-marks-delay", "Duration after which the blocks marked for deletion can be filtered out. "+
+		"If delete-delay duration provided to compactor or bucket verify component, it will upload deletion-mark.json file to mark after what duration the block should be deleted rather than deleting the block straight away. "+
```

Comment:
if.. missing `if` what exactly? (:

Reply:
We need to add more context probably. The goal of `ignore-deletion-marks-delay` is to ignore blocks with deletion-marks with some delay. This ensures store can still serve blocks-meant-to-be-deleted but does not have a replacement yet (due to object storage eventual consistency)

Reply:
I have added it like: `The idea of ignore-deletion-marks-delay is to ignore blocks that are marked for deletion with some delay. This ensures store can still serve blocks that are meant to be deleted but do not have a replacement yet.`

Resolving this, since this is addressed, feel free to re-open this or create a new thread.